golem-component-service-base/src/service/component.rs

```diff
@@ -320,10 +328,40 @@ pub trait ComponentService<Owner: ComponentOwner>: Debug {
         installed_plugins: Vec<PluginInstallation>,
         dynamic_linking: HashMap<String, DynamicLinkedInstance>,
         owner: &Owner,
-    ) -> Result<Component<Owner>, ComponentError>;
+        apply_transformations: impl AsyncFnOnce(
```

Comment:
This guy is new and is used to break the cycle with the plugin service.
Converted the trait to a struct to be able to use AsyncFn here